src/components/shared/Dialog/DialogHeading/MotionDomainSelect/MotionDomainSelect.tsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import { Id } from '@colony/colony-js';
+import { useIntl, defineMessages, MessageDescriptor } from 'react-intl';
+
+import { SelectOption } from '~shared/Fields';
+import DomainDropdown from '~shared/DomainDropdown';
+
+import { Colony } from '~types';
+
+import styles from './MotionDomainSelect.css';
+
+const displayName = 'DialogHeading.MotionDomainSelect';
+
+const MSG = defineMessages({
+  createDomain: {
+    id: `${displayName}.createDomain`,
+    defaultMessage: 'Create motion in ',
+  },
+});
+
+interface Props {
+  colony: Colony;
+  name?: string;
+  onDomainChange?: (domainId: number) => any;
+  filterDomains?: (option: SelectOption) => boolean;
+  disabled?: boolean;
+  dropdownLabel?: MessageDescriptor;
+}
+
+const MotionDomainSelect = ({
+  colony,
+  name = 'motionDomainId',
+  onDomainChange,
+  filterDomains,
+  disabled = false,
+  dropdownLabel,
+}: Props) => {
+  const { formatMessage } = useIntl();
+  const renderActiveOption = (
+    option: SelectOption | undefined,
+    label: string,
+  ) => {
+    /*
+     * @NOTE This is so that the active item is displayed as `Root/Current Domain`
+     * when a subdomain is selected
+     */
+    let displayLabel =
+      (option?.value || Id.RootDomain) === Id.RootDomain
+        ? `${formatMessage(dropdownLabel || MSG.createDomain)} ${label}`
+        : `${formatMessage(dropdownLabel || MSG.createDomain)} ${formatMessage({
```

Comment:
Tiny note, you can also use `formatText` utility which saves you having to call `useIntl` above